arangod/Aql/Condition.cpp

```diff
@@ -1233,6 +1233,63 @@ AstNode* Condition::collapse(AstNode const* node) {
   return newOperator;
 }
 
+void switchSidesInCompare(AstNode* node) {
+  // switch members of BINARY_LT/GT/LE/GE_NODES
+  // and change operator accordingly
+
+  auto first = node->getMemberUnchecked(0);
+  auto second = node->getMemberUnchecked(1);
+
+  node->changeMember(0,second);
+  node->changeMember(1,first);
+
+  switch(node->type) {
+    case NODE_TYPE_OPERATOR_BINARY_LT:
+      node->type = NODE_TYPE_OPERATOR_BINARY_GT;
+      break;
+    case NODE_TYPE_OPERATOR_BINARY_GT:
+      node->type = NODE_TYPE_OPERATOR_BINARY_LT;
+      break;
+    case NODE_TYPE_OPERATOR_BINARY_LE:
+      node->type = NODE_TYPE_OPERATOR_BINARY_GE;
+      break;
+    case NODE_TYPE_OPERATOR_BINARY_GE:
+      node->type = NODE_TYPE_OPERATOR_BINARY_LE;
+      break;
+    default:{}
```

Comment:
can we add an assert here? because it the method exchanges the two operands but the operator type is unexpected, it will silently break the result.